hack/tools/prowjob-gen/README.md

```diff
@@ -0,0 +1,68 @@
+# prowjob-gen
+
+Prowjob-gen is a tool which helps generating prowjob configuration.
+
+## Usage
+
+Flags:
+
+```txt
+  -config string
+        Path to the config file
+  -output-dir string
+        Path to the directory to create the files in
+  -templates-dir string
+        Path to the directory containing the template files referenced inside the config file
+```
+
+When running prowjob-gen, all flags need to be provided.
+The tool then will iterate over all templates defined in the config file and execute them per configured branch.
+
+The configuration file is supposed to be in yaml format and to be stored inside the [test-infra](https://github.com/kubernetes/test-infra)
+repository, we have to make sure it is not getting parsed as configuration for prow.
+Because of that the top-level key for the configuration file is `prow-ignored:`.
+
+A sample configuration looks as follows:
+
+```yaml
+prow_ignored:
+  branches:
+    main: # values below the branch here are available in the template 
+      kubekinsImage: "gcr.io/k8s-staging-test-infra/kubekins-e2e:v20231208-8b9fd88e88-1.29"
+      interval: "2h"
+      kubernetesVersionManagement: "v1.26.6@sha256:6e2d8b28a5b601defe327b98bd1c2d1930b49e5d8c512e1895099e4504007adb"
+      kubebuilderEnvtestKubernetesVersion: "1.26.1"
+      upgrades:
+      - from: "1.29"
+        to: "1.30"
+
+  templates:
+  - name: "cluster-api-periodics.yaml.tpl"
+    format: "cluster-api-periodics-%s.yaml"
```

Comment:
It would be nice if we can do something like this
```suggestion
    format: "cluster-api-periodics-{{ .branch }}.yaml"
```

This way it would be very easy to tell what gets added here. WDYT?
(not super important though)